pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/LightningGPUSimulator.cpp

```diff
@@ -0,0 +1,557 @@
+    return result;
+}
+
+void LightningGPUSimulator::ReleaseQubit(QubitIdType q) {
+    this->qubit_manager.Release(q);
+}
+
+void LightningGPUSimulator::ReleaseAllQubits() {
+    this->qubit_manager.ReleaseAll();
+    this->device_sv = std::make_unique<StateVectorT>(0); // reset the device
+}
+
+auto LightningGPUSimulator::GetNumQubits() const -> std::size_t {
+    return this->device_sv->getNumQubits();
+}
+
+void LightningGPUSimulator::StartTapeRecording() {
+    RT_FAIL_IF(this->tape_recording, "Cannot re-activate the cache manager");
+    this->tape_recording = true;
+    this->cache_manager.Reset();
+}
+
+void LightningGPUSimulator::StopTapeRecording() {
+    RT_FAIL_IF(!this->tape_recording,
+               "Cannot stop an already stopped cache manager");
+    this->tape_recording = false;
+}
+
+auto LightningGPUSimulator::CacheManagerInfo()
+    -> std::tuple<std::size_t, std::size_t, std::size_t,
+                  std::vector<std::string>, std::vector<ObsIdType>> {
+    return {this->cache_manager.getNumOperations(),
+            this->cache_manager.getNumObservables(),
+            this->cache_manager.getNumParams(),
+            this->cache_manager.getOperationsNames(),
+            this->cache_manager.getObservablesKeys()};
+}
+
+void LightningGPUSimulator::SetDeviceShots(std::size_t shots) {
+    this->device_shots = shots;
+}
+
+auto LightningGPUSimulator::GetDeviceShots() const -> std::size_t {
+    return this->device_shots;
+}
+
+void LightningGPUSimulator::SetDevicePRNG(std::mt19937 *gen) {
+    this->gen = gen;
+}
+
+/// LCOV_EXCL_START
+void LightningGPUSimulator::PrintState() {
+    using std::cout;
+    using std::endl;
+
+    const std::size_t num_qubits = this->device_sv->getNumQubits();
+    const std::size_t size = Pennylane::Util::exp2(num_qubits);
+
+    std::vector<std::complex<double>> state(size, {0.0, 0.0});
+
+    this->device_sv->CopyGpuDataToHost(state.data(), size);
+
+    std::size_t idx = 0;
+    cout << "*** State-Vector of Size " << size << " ***" << endl;
+    cout << "[";
+    for (; idx < size - 1; idx++) {
+        cout << state[idx] << ", ";
+    }
+    cout << state[idx] << "]" << endl;
+}
+/// LCOV_EXCL_STOP
+
+void LightningGPUSimulator::SetState(DataView<std::complex<double>, 1> &data,
+                                     std::vector<QubitIdType> &wires) {
+    std::size_t expected_wires = static_cast<std::size_t>(log2(data.size()));
+    RT_ASSERT(expected_wires == wires.size());
+    std::vector<std::complex<double>> data_vector(data.begin(), data.end());
+    this->device_sv->setStateVector(data_vector.data(), data_vector.size(),
+                                    getDeviceWires(wires));
+}
+
+void LightningGPUSimulator::SetBasisState(DataView<int8_t, 1> &data,
+                                          std::vector<QubitIdType> &wires) {
+    std::vector<std::size_t> basis_state(data.begin(), data.end());
+    this->device_sv->setBasisState(basis_state, getDeviceWires(wires));
```

Comment:
No action items, just leaving a note here saying the `set(basis)state` methods are correct :100:  (as per the changes made in #869 ), cc @erick-xanadu